src/Quartz/Impl/StdSchedulerFactory.cs

```diff
@@ -508,11 +516,14 @@ protected virtual void ValidateConfiguration()
 
             Type tpType = loadHelper.LoadType(cfg.GetStringProperty(PropertyThreadPoolType)) ?? typeof(SimpleThreadPool);
 
+#if WINDOWS_THREADPOOL  // TODO (NetCore Port) - We can't use the ClrThreadPool on .NET Core because cross-plat threadpool implementations don't expose all the APIs we need
+                        // Use the SimpleThreadPool there for now and in the future create a new async-friendly threadpool that will work cross-plat.
             if (tpType == typeof(SimpleThreadPool))
```

Comment:
@ericeil, I'm helping to port the Quartz.Net scheduling library to Core and one of the obstacles is ThreadPool changes. The Quartz.Net library allows users to schedule tasks and also control the level of concurrency in the task runners that process those schedules and kick things off.

One of the ways this was done previously was by queuing tasks in the ThreadPool when their schedules triggered while using GetMaxThreads/SetMaxThreads/GetAvailableThreads to make sure that the number of tasks running at any particular time was capped at the limit set by the user.

So, two questions here -

1. Can you help us understand briefly what the rationale was for removing those APIs? Was it more a matter of what was practical to implement cross-plat or was it more that those APIs just aren't recommended for use anymore?
2. How would you suggest we make a library like this work in .NET Core? It seems like it should be possible to create a task-based solution that begins tasks at their trigger times while keeping a count of executing tasks so that we limit concurrency. All that could be done without using the CLR's thread pool. Do you think that's the best approach?

Thanks for letting us pick your brain!